Sources/App/Controllers/SearchController.swift

```diff
@@ -19,15 +19,26 @@ import Vapor
 
 struct SearchController {
 
-    func show(req: Request) throws -> EventLoopFuture<HTML> {
+    func show(req: Request) async throws -> HTML {
+                
         let query = req.query[String.self, at: "query"] ?? ""
         let page = req.query[Int.self, at: "page"] ?? 1
-        return API.search(database: req.db,
+        
+        let response = try await API.search(database: req.db,
                           query: query,
                           page: page,
-                          pageSize: Constants.resultsPageSize)
-            .map { SearchShow.Model.init(page: page, query: query, response: $0) }
-            .map { SearchShow.View.init(path: req.url.path, model: $0).document() }
+                          pageSize: Constants.resultsPageSize).get()
+        
+        let matchedKeywords = response.results.compactMap { $0.keywordResult?.keyword }
```

Comment:
I don’t think that’ll matter, because the query already short circuits to an empty array, doesn’t it?

Anyway, I just wanted to leave a note to have a quick think about it. I don’t pages > get called a lot, so this isn’t a huge deal tbh. Just “feels” a bit off to do it on every page when it’s not necessary!